src/file.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import * as fsExtra from 'fs-extra';
+import AdmZip from 'adm-zip';
+import zlib from 'zlib';
+import tarStream from 'tar-stream';
+import fs from 'fs';
+import path from 'path';
+import { SONAR_CACHE_DIR } from './constants';
+import { log, LogLevel } from './logging';
+
+export async function getCachedFileLocation(md5: string, filename: string) {
+  const filePath = path.join(SONAR_CACHE_DIR, md5, filename);
```

Comment:
We are not supporting `sonar.userHome` for now, I would create a subtask for that or make it part of the scope of this PR, if that sounds good

Reply:
it makes sense, we can review all cache scenarios in this ticket to make sure we have proper support
https://sonarsource.atlassian.net/browse/SCANNPM-20